File: atcoder.jp/contests/abc134/tasks/abc134_b/main.hs
main :: IO ()
main = do
  [n, d] <- getIntList
  let ans = n `divCeil` (2 * d + 1)
  print ans

getIntList :: IO [Int]
getIntList = map read . words <$> getLine

divCeil :: Integral a => a -> a -> a
a `divCeil` b = (a + b - 1) `div` b

File: atcoder.jp/contests/abc134/tasks/abc134_b/main.cpp
#include <bits/stdc++.h>
using namespace std;

int div_ceil(int a, int b) { return (a + b - 1) / b; }

int main() {
  int N, D;
  cin >> N >> D;

  int ans = div_ceil(N, 2 * D + 1);

  cout << ans << endl;
}
